mobile/src/screens/note/note.tsx

```diff
@@ -0,0 +1,153 @@
+import { type RouteProp, useRoute } from '@react-navigation/native';
+import React from 'react';
+
+import {
+  BackHandler,
+  FormController,
+  LinearGradient,
+  ScrollView,
+  TextInput,
+  View,
+} from '#libs/components/components';
+import { debounce, sanitizeInput } from '#libs/helpers/helpers';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useMemo,
+} from '#libs/hooks/hooks';
+import { type JournalNavigationParameterList } from '#libs/types/types';
+import { type JournalEntryCreateRequestDto } from '#packages/journal/journal';
+import { actions as journalActions } from '#slices/journal/journal';
+
+import {
+  DEFAULT_NOTE_PAYLOAD,
+  NOTE_SANITIZER_OPTIONS,
+  SAVE_NOTE_TIMEOUT,
+  TEXT_PLACEHOLDER,
+} from './libs/constants';
+import { styles } from './styles';
+
+type NoteScreenRouteProperty = RouteProp<
+  JournalNavigationParameterList,
+  'Note'
+>;
+
+const handleSanitizeInput = (textInput: string): string => {
+  return sanitizeInput(textInput, NOTE_SANITIZER_OPTIONS);
+};
+
+const Note: React.FC = () => {
+  const route = useRoute<NoteScreenRouteProperty>();
+  const id = route.params.id;
+
+  const dispatch = useAppDispatch();
+
+  const { selectedJournalEntry } = useAppSelector(({ journal }) => {
+    return {
+      selectedJournalEntry: journal.selectedJournalEntry,
+    };
+  });
+
+  const { control, watch, isDirty } = useAppForm({
+    defaultValues: selectedJournalEntry
+      ? {
+          title: selectedJournalEntry.title || DEFAULT_NOTE_PAYLOAD.title,
+          text: selectedJournalEntry.text || DEFAULT_NOTE_PAYLOAD.text,
+        }
+      : DEFAULT_NOTE_PAYLOAD,
+    mode: 'onChange',
+  });
+
+  const { title: titleValue, text: textValue } = watch();
+
+  const handleSaveNote = useCallback(
+    (data: JournalEntryCreateRequestDto) => {
+      void dispatch(
+        journalActions.updateJournalEntry({
+          id: Number(id),
+          title: data.title || DEFAULT_NOTE_PAYLOAD.title,
+          text: data.text,
+        }),
+      );
+    },
+    [id, dispatch],
+  );
+
+  const handleSaveNoteWithDebounce = useMemo(() => {
+    return debounce(handleSaveNote, SAVE_NOTE_TIMEOUT);
+  }, [handleSaveNote]);
+
+  useEffect(() => {
+    if (id && isDirty) {
+      handleSaveNoteWithDebounce({
+        title: titleValue,
+        text: textValue,
+      });
+    }
+  }, [titleValue, textValue, handleSaveNoteWithDebounce, id, isDirty]);
+
+  useEffect(() => {
+    const handleBeforeBackNavigation = (): boolean => {
+      handleSaveNote({
+        title: titleValue,
+        text: textValue,
+      });
+
+      return true;
```

Comment:
Do we really need this return statement?

Reply:
@fshabanov returning true tells React Native that we consumed the back button press event and it should not perform its default navigation action. if we want to navigate back without saving the note (which is not right for this app functionality) when the back button is pressed, we can remove the return true statement.  But If we want to save the note before navigating back, then keeping return true is appropriate.